src/main/java/org/opensearch/knn/index/query/KNNWeight.java

```diff
@@ -82,13 +68,17 @@ public class KNNWeight extends Weight {
 
     private final NativeMemoryCacheManager nativeMemoryCacheManager;
     private final Weight filterWeight;
+    private final ExactSearcher exactSearcher;
+
+    private static ExactSearcher DEFAULT_EXACT_SEARCHER;
```

Comment:
This is redundant , we are having two variable for ExactSearcher just to save DI , why should we not considering lazy intialization of ExactSearcher  by Singleton and get Instance here ..

Reply:
I can change to that. I guess testing and the fact that ExactSearcher has dependency on ModelDao

Reply:
I might defer changing this to later after I make a change to the ModelUtil to get modelMetadata properly